### ISSUE TITLE
Memory usage of TextDataBunch(DataBunch).from_ids - ToIntsProcessor

### ISSUE THREAD

**Issue by kasparlund** [fastai/fastai#1357]
**context**: i am working on a notebook with all the steps to train a language-model based on wikipedia in any language using sentencepiece and fastai. I am also trying to push the limits of the size of the training data. The objective it to that the lasts english wikipedia dump 16 gb and only remove short sentences. This is a good test of the memory consumption i the fastai textmodule.

**The problem:** The lm api has a hugh memory consumption

the factory method from_ids call/allocates  ToIntsProcessor() that convert ids to dtype=np.int64.

With a vocab of 32000 the ids stays within the range of int16
With a vocab of 64000 the ids stays by far within the range of int32 og uint16

making the conversion increases memory usage as follows:
-the copy increases it by a factor 2
-going from int16 to int 64 increases it by a factor 4

That is for a source of int16 ids the conversion increases memory consumption by a factor of 8. When the garbage collector sets in we get back to a factor 4. making a conversion when feeding the model is relatively cheep compared to the downside and the memory consumption of the current approach.

**Solution:**
Basically ToIntsProcessor should only raise an exception if the user provides anything but numbers

**Comment by sgugger** [comment]
Should be fixed by [this commit](https://github.com/fastai/fastai/commit/95097a408944094626a6c93da33ea8eba5edde1f). We only operate the conversion before collating for a batch.
